tests/behat/features/bootstrap/BookContext.php

```diff
@@ -72,6 +72,23 @@ public function createTopics(TableNode $topicsTable) : void {
     }
   }
 
+  /**
+   * @Given Book structure is enabled for topics
+   */
+  public function enableBookStructureForTopics() : void {
+
+    $config = \Drupal::configFactory()->getEditable('book.settings');
+
+    if ($config->isNew()) {
+      throw new \Exception("The book.settings configuration did not yet exist, is the 'book' module enabled?");
+    }
+
+    $allowed_types = $config->get('allowed_types');
+    $allowed_types[] = 'topic';
+
+    $config->set('allowed_types', $allowed_types)->save();
+  }
```

Comment:
We can make a slight tweak here to make this generic. Even though we don't use it yet, it doesn't complicate the code more but provides us some options in the future :D 

```suggestion
  /**
   * Enable Drupal core book functionality for a content type.
   *
   * @Given book structure is enabled for :content_type
   */
  public function enableBookStructureForContentType(string $content_type) : void {

    $config = \Drupal::configFactory()->getEditable('book.settings');

    if ($config->isNew()) {
      throw new \Exception("The book.settings configuration did not yet exist, is the 'book' module enabled?");
    }

    $allowed_types = $config->get('allowed_types');
    $allowed_types[] = $content_type;

    $config->set('allowed_types', $allowed_types)->save();
  }
```

I've lowercased the b in "Book" to make it a proper English sentence and I've put "topics" behind "$content_type" so we can also use it e.g. for "event" if the need arises.

I like the exception you've written for when the `book.settings` config doesn't exist, it helps developers find a good next step! :D

Reply:
Added